p2p/config.go

```diff
@@ -17,6 +17,9 @@ type Config struct {
 	BootstrapPort string `mapstructure:"bootstrap_port" json:"bootstrap_port,omitempty"`
 	// Use STUN protocol for public addr discovery
 	UseStun bool `mapstructure:"use_stun" json:"use_stun,omitempty"`
+	// a driver-specific data source name, usually consisting of
+	// at least a database name and connection information.
+	DataSourceName string `mapstructure:"data_source_name" json:"data_source_name,omitempty"`
```

Comment:
Hint: `supernode` must also run without configuration.

Reply:
I guess `DataSourceName` should be called `DataDir`.

`DataDir` by default should has the path  `work-dir`/p2p, e.g. [example](https://github.com/pastelnetwork/gonode/blob/fcf89eae0d2a649caee1adbd6384940513d4b2b8/supernode/cmd/app.go#L129). Use `ioutil.TempDir` for unit tests